okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/UrlSelectorImpl.java

```diff
@@ -54,6 +57,16 @@ private UrlSelectorImpl(ImmutableList<HttpUrl> baseUrls, Duration failedUrlCoold
         Preconditions.checkArgument(!failedUrlCooldown.isNegative(), "Cache expiration must be non-negative");
     }
 
+    private <T> List<T> randomize(List<T> list, boolean randomizeOrder) {
+        if (randomizeOrder) {
+            List<T> shuffledList = new ArrayList<>(list);
+            Collections.shuffle(shuffledList);
+            return Collections.unmodifiableList(shuffledList);
+        } else {
+            return list;
+        }
+    }
+
     /**
      * Creates a new {@link UrlSelector} with the supplied URLs. The order of the URLs may be randomized by setting
```

Comment:
update the docs to indicate URLs are randomized every 10 minutes when randomizeOrder is true

Reply:
also worth noting that except for testing we expect randomizeOrder to generally be set to true